### PR TITLE
feat(auth): Added form validations using Zod schema

### PR DESCRIPTION
**Pull request by Ibinola**
### Description

Added form validations using Zod schema for all inputs of user workflows.

### Related Issue

Fixes - [#96](https://github.com/trypear/pear-landing-page/issues/96)

### Changes Made

- added a loading state when submitting forms
-  error handling to display validation messages on invalid inputs.

### Screenshots

Sign In Page
image

Sign Up Page
image

Reset Password
image

### Checklist

- [x] I have tagged the issue in this PR.
- [x] I have attached necessary screenshots.
- [x] I have provided a short description of the PR.
- [x] I ran `yarn build` and build is successful

### CODE REVIEW

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts]: For the ErrorMessages type consider using a mapped type for better maintainability
```suggestion
export type ErrorMessages = {
  [key in keyof SignUpFormData | 'form']?: string;
};
```

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts]:
```suggestion
    email: emailSchema,
```

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts]:
```suggestion
    email: emailSchema
```

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts:6]: max length also here
```suggestion
  company_name: z.string()
  .min(1, { message: "Company name is required." })
  .max(100, { message: "Company name is too long." }),
```

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts]:
```suggestion
  email: emailSchema,
```

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts]: Consider adding max length validations to prevent excessively long inputs eg.
```suggestion
  full_name: z.string().min(1, { message: "Name is required." }).max(100, { message: "Name is too long." }),
```

File: utils/form-schema.ts
Comment (ICEPrey) [inline utils/form-schema.ts:2]: For consistency we could create reusable schemas for common fields like email
```suggestion
const emailSchema = z.string().email({ message: "Email address is invalid." });
```

**Comment by Fryingpannn** [comment]
WILL REVIEW THIS SOON. Haven't checked yet but wanted to flag that there is some built in validations rn

e.g.
image

**Review by shrix1 — CHANGES_REQUESTED** [review]
Can you please follow the shadcn form
https://ui.shadcn.com/docs/components/form

Screenshot 2024-06-27 at 10 24 07 AM

**Comment by ICEPrey** [comment]
> Can you please follow the shadcn form https://ui.shadcn.com/docs/components/form
> Screenshot 2024-06-27 at 10 24 07 AM

yea I do agree that we should stick to shadcn components and use Form

File: package.json
Comment (ICEPrey) [inline package.json:26]:
```suggestion
    -"@shadcn/ui": "^0.0.4",
```
**Remove @shadcn/ui package**
shadcn/ui is not a traditional package that we install via npm or yarn. Instead it's a collection of reusable components that we add to our project individually. So when you run `npx shadcn-ui@latest add input` The CLI adds the Input component files directly to your project in the components/ui directory. It shouldn't install a shadcn/ui package in your node_modules

**Review by shrix1 — COMMENTED** [review]
LGTM , nice job @Ibinola Thanks for refactoring

**Comment by Ibinola** [comment]
> > > * Build failed Please check `yarn build` on local.
> > > * Other than that code LGTM , nice job @Ibinola Thanks for refactoring
> > 
> > 
> > fixed, also noticed when I had the optional method on the company name field on the sign-up form, it returned `undefined` image
> 
> * CompanyName should be optional
> * maybe do like `data?.company_name || null`

ohh okay i guess, yes did something like that, still didn't work, but this worked `data.company_name || ""`

**Review by ICEPrey — COMMENTED** [review]
Looks good to me, need another maintainer to look at my commits before approval though.

**Review by Fryingpannn — APPROVED** [review]

**Pull request merged**
